File: hello-impl/src/main/scala/com/example/hello/impl/HelloServiceImpl.scala
/*
 * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
 */

package com.example.hello.impl

import com.example.hello.api.HelloService
import com.lightbend.lagom.scaladsl.api.ServiceCall

import scala.concurrent.Future

/**
  * Implementation of the HelloService.
  */
class HelloServiceImpl() extends HelloService {

  override def hello(id: String) = ServiceCall { _ =>
    Future.successful(s"Hi $id!")
  }
}

File: grpc-example/java-api/hello-impl/src/main/java/com/example/hello/impl/HelloServiceImpl.java
package com.example.hello.impl;

import akka.NotUsed;
import com.example.hello.api.HelloService;
import com.lightbend.lagom.javadsl.api.ServiceCall;

import javax.inject.Inject;
import java.util.concurrent.CompletableFuture;

public class HelloServiceImpl implements HelloService {

    @Inject
    public HelloServiceImpl(){

    }

    @Override
    public ServiceCall<NotUsed, String> hello(String id) {
        return req -> CompletableFuture.completedFuture("Hi " + id + "!");
    }
}
